eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -549,7 +567,14 @@ object Peer {
   case object CONNECTED extends State
 
   case class Init(previousKnownAddress: Option[InetSocketAddress], storedChannels: Set[HasCommitments])
-  case class Connect(uri: NodeURI)
+  case class Connect(nodeId: PublicKey, address_opt: Option[HostAndPort]) {
```

Comment:
Why didn't you use the `Reconnect` object here? Instead of a `Connect` without address?

Reply:
Because logically a connection to a peer does not necessarily require knowledge of his address, eclair can support connect-to-pubkey (with an optional address).